tests/system/test_monitoring.py

```diff
@@ -0,0 +1,40 @@
+from apmserver import integration_test
+from apmserver import ElasticTest
+
+import urlparse
+
+
+@integration_test
+class Test(ElasticTest):
+    def config(self):
+        cfg = super(Test, self).config()
+
+        # Extract username/password from the URL, and store
+        # separately from elasticsearch_host. This is necessary
+        # because Beats will use the userinfo from the URL in
+        # preference to specified values.
+        url = urlparse.urlparse(cfg["elasticsearch_host"])
+        if url.username:
+            cfg["elasticsearch_username"] = url.username
+        if url.password:
+            cfg["elasticsearch_password"] = url.password
+        bare_netloc = url.netloc.split('@')[-1]
+        url = list(url)
+        url[1] = bare_netloc
+
+        # Set a password for the built-in apm_system user, and use that for monitoring.
+        monitoring_password = "changeme"
+        self.es.xpack.security.change_password(username="apm_system",
+                                               body='{"password":"%s"}' % monitoring_password)
```

Comment:
By default tests are run with a non-admin user. This request probably requires admin rights. You could run it using something along the lines of
```
admin_user = os.getenv("ES_SUPERUSER_USER", "admin")
admin_password = os.getenv("ES_SUPERUSER_PASS", "changeme")
admin_es = Elasticsearch([self.get_elasticsearch_url(admin_user, admin_password)])
```

Reply:
Thanks, fixed.